internal/config/validation.go

```diff
@@ -549,6 +549,16 @@ func validInstanceTypeForProvider(insType string, attestation variant.Variant) b
 			}
 		}
 	case variant.QEMUVTPM{}, variant.QEMUTDX{}:
+		// only allow confidential instances on stackit cloud using QEMU vTPM
+		if cloud := provider.OpenStack.Cloud; strings.ToLower(cloud) == "stackit" {
```

Comment:
We need an `if provider.OpenStack != nil` here, as there's more scenarios where we'll have `case variant.QEMUVTPM{}, variant.QEMUTDX{}`, such as MiniConstellation, and else we'd run into a nil panic there here when dereferencing `provider.OpenStack`.

We might also want "success" test case for these variants without the OpenStack / STACKIT case here.

Reply:
I've added a provider check and two test cases. Although I'm not entirely sure which provider Miniconstellation is using, I made an educated guess: `ProviderConfig{QEMU: &QEMUConfig{}}` with the variant `variant.QEMUTDX{}`.